README.md

```diff
@@ -3,21 +3,163 @@
 ## Scripts
 
 ### hh:compile
+
 Compiles the contracts with Hardhat
 
 ### hh:deploy
+
 Runs the deployment script with the network set in `process.env.NETWORK`.
 
 ### hh:node
+
 Starts a local hardhat node with the `localhost` network.
 
 ## Contracts
-Contracts are located in the `/contracts` folder.
+
+There are 5 main types of contracts along with supporting solidity libraries, which are located in the `/contracts` folder.
+
+1. Token `Token.sol`
+2. Vault `Vault.sol`
+3. Market `Market.sol` and `MarketCurved.sol`
+4. Oracle `MarketOracle.sol`
+5. Registry `Registry.sol`
+
+### Token
+
+Horse Link issues 100m standard ERC20 tokens HL / Horse Link for its members to be used in the future as a DAO governance token, distribution of protocol fees and other member perks.
+
+### Vaults
+
+The Vault contracts are ERC4626 contracts used to manage the underlying ERC20 assets of the LP Providers. They are used to store the assets of the users and to allow them to deposit and withdraw assets that get lent to Market contracts for a fee which is agreed upon in the `setMarket` function. The users are minted a ERC20 share that represents their share of the underlying assets in the Vault.
+
+The following is a worked example on how the relationship between users' deposits and their shares work.
+
+1. Alice deposits 1000 DAI into the Vault and receives 1000 shares.
+2. Bob deposits 1000 DAI into the Vault and receives 1000 shares.
+
+| User  | Action  | Amount   | Shares | Total Assets | Total Shares |
+| ----- | ------- | -------- | ------ | ------------ | ------------ |
+| Alice | Deposit | 1000 DAI | 1000   | 1000 DAI     | 1000         |
+| Bob   | Deposit | 1000 DAI | 1000   | 2000 DAI     | 2000         |
+
+The Vault is now holding 2000 DAI in "totalAssets". If Alice withdraws 500 shares, she will receive 500 DAI. Now, lets say a bet of 1800 DAI is placed on a market that is backed by the Vault at 1:1. The Vault will lend 1800 DAI to the market and have a total assets of 200 DAI and total exposure of 1800 DAI.
+
+3. Vault lends 1800 DAI to the market and has 200 DAI left in total assets.
+
+| User  | Action  | Amount   | Shares | Total Assets | Total Shares |
+| ----- | ------- | -------- | ------ | ------------ | ------------ |
+| Alice | Deposit | 1000 DAI | 1000   | 1000 DAI     | 1000         |
+| Bob   | Deposit | 1000 DAI | 1000   | 2000 DAI     | 2000         |
+| Vault | Lend    | 1800 DAI | 0      | 200 DAI      | 2000         |
+
+While the bet is active, Alice can redeem her 1000 shares if she chooses so, but will but the exchange rate per share will be significantly lower than 1 DAI per share. Her share is represented by the following equation:
```

Comment:
@bazmatic @bitcoinbrisbane @Mitchell139 is this still the case given recent changes?

Reply:
@MikeyRobertson